### ISSUE TITLE
ADAM does not accept keyword arguments

### ISSUE THREAD

**Issue by jessebett** [FluxML/Flux.jl#871]
`?Flux.ADAM` gives the help string `ADAM(η = 0.001, β = (0.9, 0.999))`

But if you try to create an optimizer with those (unicode) keyword arguments then you get a method error for unexpected arguments.

**Comment by MikeInnes** [comment]
Unfortunately this is a notation clash between [named] default arguments and keyword arguments. In signatures it's unambiguous because you have to use `;` for kwargs, but this doesn't work when you actually call a function.

How about writing the signature as

```julia
ADAM(η, β)
ADAM() => ADAM(0.001, (0.9, 0.999))
```

**Comment by jessebett** [comment]
Yes I think the help string mislead me to think those were keyword arguments, but you're right that I should have looked for the `;`.

**Comment by ToucheSir** [comment]
Can we call this closed by https://github.com/FluxML/Flux.jl/pull/904 and subsequent PRs?

**Comment by darsnack** [comment]
The initial issue is the same, but that's because it is the standard way to document this situation in a docstring. I agree that having to look for `;` isn't helpful. I am pushing for only LR as a positional argument in Optimisers.jl (which would make this a non-issue).
